src/core/agent_space_interaction.jl

```diff
@@ -113,20 +113,40 @@ end
 #######################################################################################
 # Adding agents
 #######################################################################################
+# TODO: the Source code of `add_agent!` for discrete space is overly complex.
```

Comment:
no, this is still valid. The source code wasn't changed at all, only the docstrings were. the source still uses the complicated format where `add_agent_pos!` calls `add_agent!` instead of the other way around.